hera_qm/xrfi.py

```diff
@@ -328,64 +311,76 @@ def detrend_channel_diff_filt(data, noise, nsig=6, kernel_widths=[1, 2, 4], wgts
-        for t in range(data.shape[0]):
-            _d = np.convolve(dw[t], kernel, mode="same") - dw[t]
-            _w = np.convolve(wgts[t], kernel, mode="same") - wgts[t]
-            residual[t] = dw[t] - _d / _w.clip(1, np.inf)
-            wgts[t] = np.where(residual[t] > noise[t] * kernel_std * nsig, 0, wgts[t])
+        # Estimate noise level in absence of RFI
+        sigma = np.abs(model) * (noise / data)
+        res = data - model
 
-    return wgts
+        # Identify outlier channels
+        wgts = np.where(res > sigma * nsig, 0., 1.)
 
+    return np.isclose(wgts, 0)
 
-auto_flaggers = {'detrend_dpss_filt': detrend_dpss_filt, 'detrend_channel_diff_filt': detrend_channel_diff_filt}
+auto_flaggers = {'dpss_flagger': dpss_flagger, 'channel_diff_flagger': channel_diff_flagger}
 
 
-def flag_autos(data, flag_method="detrend_diff_filt", nsig=6, int_count=None, antenna_class=None, wgts=None,
-               array_wgts_threshold=0.2, **flag_kwargs):
+def flag_autos(data, flag_method="channel_diff_flagger", nsig=6, int_count=None, antenna_class=None, flags={},
+               flag_threshold=0.1, **flag_kwargs):
     """
     Driver function for identifying RFI in auto-correlations. Flags auto-correlations on a per antenna basis using
-    a given function and consolidates the weights into a single array-wide weights array. Return weights 
+    a given function and consolidates the weights into a single array-wide weights array. Returns a dictionary of 
+    boolean flag arrays and an array averaged flag array where channels flagged in "flag_threshold" fraction of antennas
+    will be flagged in the array averaged flag array.
 
     Parameters:
     ----------
     data: DataContainer
         DataContainer containing numpy arrays of auto-correlation data with shape (ntimes, nfreqs)
-    flag_method: str, default='detrend_diff_filt'
-        Method used for flagging auto-correlations. Current options are 'dpss_flagger' and 'detrend_channel_diff_filt'
+    flag_method: str, default='channel_diff_flagger'
+        Method used for flagging auto-correlations. Current options are 'dpss_flagger' and 'channel_diff_flagger'
     nsig: float, default=6
         The number of sigma in the metric above which to flag pixels.
     int_count: float, default=None
         Number of samples per integration in correlator (dt * dnu)
     antenna_class: AntennaClassification, default=None
         Optional AntennaClassification object. If provided, the flagging method chosen will skip antennas marked "bad"
-    wgts: dictionary or DataContainer: default=None
-        Dictionary or DataContainer containing data weights. If wgts=None, then weights will be set to ones
-    array_wgts_threshold: float, default=0.2
+    flags: dictionary or DataContainer: default={}
+        Dictionary or DataContainer containing boolean flag arrays. If flags=None, then no data will be masked
+    flag_threshold: float, default=0.1
```

Comment:
This variable name is a bit misleading, at least to me. I'd call it flag_broadcast_thresh or similar.

Reply:
Changed the name of this to `flag_broadcast_thresh`